### DISCUSSION TITLE
temperature dependency on chemistry

### DISCUSSION THREAD

**Discussion by EMKOUTAIBA** [firemodels/cfast#2110] [Q&A]
Hi everyone,

We’re trying to run a simple case of a compartment fire. We are investigating the influence of different parameters on the upper layer temperature. However, we are slightly perturbed by the significant impact of the chemical composition (particularly the amount of oxygen, n_O) on the calculated constrained heat release rate (hrr_constrained). I have attached two cases where we quadrupled the amount of n_O in the combustible, which resulted in almost doubling the upper layer temperature.
This substantial dependency seems to stem from the o2_entrained, which depends on the entrainment_rate and source_o2 (nu_o2). Could you please provide some insights to help clarify these behaviors?
[Cfast_GitHub.zip](https://github.com/firemodels/cfast/files/15431023/Cfast_GitHub.zip)

**Comment by drjfloyd** [comment]
Most ordinary combustibles release ~13000 kJ/kg of O2 consumed from the air. DId you adjust your fuel heat of combustion to maintain this value as you changed the amount of O2 in the fuel? If not, then you had non-physical values of the fuel heat of combustion which is going to lead to non-physical results.

**Reply by tGagnonCorvid** [reply]
Could you speak towards that process of adjusting the heat of combustion for the O2 component of the fuel chemistry. I did some earlier sims where I was trying to disable the lower oxygen limit by setting the fuel chemistry to H1 O99 or something similar (my though process was that this would give almost any fire excess oxygen as to follow the prescribed HRR curve regardless of O2 availability). This did not end up working, but I left the O99 in my input file as I assumed it didn't really do anything. Based on what you've said above, it does not seem like this change is without consequence. Could you please elaborate on how the O99 is factoring into the sim results and how I might go about selecting a more appropriate value. For example, if my fuel is ethanol is there any alteration that needs to be done to the heat of combustion of ~26800 kJ/kg given the 1 Oxygen contained in its chemical formula of C2H6O?

**Reply by drjfloyd** [reply]
Write out the chemical equation for the combustion of 1 mole ethanol. That will give you the number of moles of oxygen from the air needed for combustion of 1 mole. Convert both over to kg.  Now you have some amount in kg of ethanol that needs some amount in kg of oxygen. You can convert the fuel heat of combustion to the oxygen heat of combustion.
